### ISSUE TITLE
Warn about missing variables in templates

### ISSUE THREAD

**Issue by voltechs** [hashicorp/terraform#6822]
Hi there,

I strongly suggest that terraform warn about missing variables in templates. It can get unwieldy to micromanage and check manually if every variable is present and being passed. The summary during a `terraform plan` is close, but it still requires manual inspection to ensure all variables are present. This is a task machines excel at.
### Terraform Version

Terraform v0.6.16
### Affected Resource(s)
- template_file
### Terraform Configuration Files

``` hcl
resource "template_file" "foo-service" {
    template = "${file("terraform/task-definitions/foo-service.json")}"

    vars {
        docker_images = "${var.docker_images.foo_service}"
        # environment = "${var.environment}"
    }
}
```

``` ccl
[
  {
    "name": "foo-service",
    "image": "${docker_images.foo_service}",
    "memory": 512,
    "essential": true,
    "portMappings": [
      {
        "hostPort": 80,
        "containerPort": 80,
        "protocol": "tcp"
      }
    ],
    "environment": [
      {
        "name": "RACK_ENV",
        "value": "${environment}"
      }
    ],
    "logConfiguration": {
      "logDriver": "json-file",
      "options": null
    }
  }
]
```
### Debug Output

N/A
### Panic Output

N/A
### Expected Behavior

Warn about missing variable.
### Actual Behavior

No warning about missing variable
### Steps to Reproduce
1. Use code listed
2. `terraform plan`
### Important Factoids

N/A
### References

N/A

**Comment by sl1pm4t** [comment]
This appears to be fixed in 0.7:

I get an error when I run your example.
```
Error applying plan:

1 error(s) occurred:

* template_file.foo-service: failed to render : 17:22: unknown variable accessed: environment
```

**Comment by dmikalova** [comment]
This isn't fixed with the 0.7 and the new data templates, maybe it is for the deprecated resource templates? Terraform doesn't even warn that the variable isn't available within that modules scope.

**Comment by mitchellh** [comment]
This should work in 0.8 and properly error.
